src/derive.md

```diff
@@ -0,0 +1,43 @@
+# Derive
+
+<!-- TODO: Maybe link to https://doc.rust-lang.org/book/appendix-03-derivable-traits.html ?
+Should the reference say more about which traits are derivable?
```

Comment:
This TODO should be its own issue. It's separate from the rest of this. The list of derivable traits should indeed be in the reference standalone from the book.

Reply:
Filed #540